_posts/2020-06-12-solidity-0.6-inheritance.md

```diff
@@ -0,0 +1,130 @@
+            x = _x;
+        }
+    }
+
+    contract B {
+        uint public y;
+        function setValue(uint _y) public {
+            y = _y;
+        }
+    }
+
+    contract C is A, B {
+    }
+
+For example in contract `C` above calling `setValue` is calling the most derived implementation from contract `B`. However this is non-obvious from the implementation.
+
+With 0.6 for the example above the compiler will now raise a type error: `Derived contract must override function "setValue". Two or more base classes define functions with the same name and parameter types`. This forces us to rething which implementation to use and explicitely override the conflicting base ones:
+
+    pragma solidity ^0.6.10;
+    contract A {
+        uint public x;
+        function setValue(uint _x) public virtual {
+            x = _x;
+        }
+    }
+
+    contract B {
+        uint public y;
+        function setValue(uint _y) public virtual {
+            y = _y;
+        }
+    }
+
+    contract C is A, B {
+        function setValue(uint _x) public override(A,B) {
+            A.setValue(_x);
+        }
+    }
+
+You can only override functions if they are marked `virtual`. Additionally any overriding function must be marked `override`. If that again should be overridable, it has to also be marked `virtual`.
+
+Note: Interface functions are implicitely virtual so when implementing an interface you have to explicitely override its functions in the implementation. There is an ongoing discussion on this design [here](https://github.com/ethereum/solidity/issues/8281).
+
+## Abstract contracts
+With version 0.5 contract that doesn't implement all of its functions is implicitely made abstract by the compiler:
+    
+    pragma solidity ^0.5.17;
+    contract X {
+        function setValue(uint _x) public virtual;
+    }
+
+With 0.6 this differentiation has to be explicit with the compiler generating error `contract X should be made abstract` 
+    
+    pragma solidity ^0.6.10;
+    abstract contract X {
+        function setValue(uint _x) public virtual;
+    }
+
+## Interfaces can inherit
+This functionality is new to version 0.6 and allows interface inheritance. The resulting interface is a superset of all inherited interfaces' functions which the contract has to implement.
+
+    pragma solidity ^0.6.10;
+    interface X {
+        function setValue(uint _x) external;
+    }
+
+    interface Y is X {
+        function getValue() external returns (uint);
+    }
+
+    contract Z is Y {
+        uint x;
+        function setValue(uint _x) external override { x = _x; }
+        function getValue() external override returns (uint) { return x; }
+    }
+
+Note that if the contract does not implement all functions it has to be marked as `abstract`.
+
+    pragma solidity ^0.6.10;
+    abstract contract Z is Y {
+        uint x;
+        function setValue(uint _x) external override { x = _x; }
+    }
+
+## Public variables can override external functions
```

Comment:
If this is not new, we can also just drop it.

Reply:
I've rewritten it to position as an improved feature with some anti patterns from 0.5. See if you like the new version.